src/logic/map_objects/tribes/tribe_basic_info.cc

```diff
@@ -125,17 +124,25 @@ std::vector<TribeBasicInfo> get_all_tribeinfos() {
 		log_err("No tribe infos found at 'tribes/initialization/<tribename>/init.lua'");
 	}
 
-	for (const auto& pair : AddOns::g_addons) {
-		if (pair.first.category == AddOns::AddOnCategory::kTribes && pair.second) {
-			const std::string dirname = kAddOnDir + FileSystem::file_separator() +
-			                            pair.first.internal_name + FileSystem::file_separator() +
-			                            "tribes";
-			if (g_fs->is_directory(dirname)) {
-				for (const std::string& tribe : g_fs->list_directory(dirname)) {
-					const std::string script_path = tribe + FileSystem::file_separator() + "init.lua";
-					if (g_fs->file_exists(script_path)) {
-						tribeinfos.push_back(Widelands::TribeBasicInfo(lua.run_script(script_path)));
-					}
+	std::vector<AddOns::AddOnInfo> addons;
+	if (addons_to_consider) {
+		addons = *addons_to_consider;
```

Comment:
Wouldn't work because it's filled in with autodetected custom content below if `addons_to_consider` is `nullptr`

Reply:
This would work:
```
        const std::vector<AddOns::AddOnInfo>* addons;
	std::vector<AddOns::AddOnInfo> enabled_tribe_addons;
	if (addons_to_consider) {
		addons = addons_to_consider;
	} else {
		for (const auto& pair : AddOns::g_addons) {
			if (pair.first.category == AddOns::AddOnCategory::kTribes && pair.second) {
				enabled_tribe_addons.push_back(pair.first);
			}
		}
		addons = &enabled_tribe_addons;
	}
```